### PR TITLE
Backport a fix to prevent failures in the aws policy

### PR DESCRIPTION
**Pull request by tas50**
We applied this already to the server repo to prevent failures

### CODE REVIEW

File: core/mondoo-aws-security.mql.yaml [hunk 1]
```diff
@@ -2264,10 +2264,7 @@ queries:
     impact: 50
     filters: |
       asset.platform == "aws-sagemaker-notebookinstance"
-    mql: |
-      aws.sagemaker.notebookInstance
-        .where(asset.name == name)
-        .all(details.kmsKey != empty)
+    mql: aws.sagemaker.notebookinstance.details.kmsKey != empty
```
Comment (mm-weber) [inline core/mondoo-aws-security.mql.yaml:2267]:
```suggestion
    mql: |
      aws.sagemaker.notebookInstances
        .where(asset.name == name)
        .all(details.kmsKey != empty)
```

Comment (mm-weber) [reply core/mondoo-aws-security.mql.yaml:2267]: Think the sagemaker resource is still broken, BUT the work-around check was also wrong:

image

**Review by mm-weber — COMMENTED** [review]
Workaround is still needed on `cnspec 9.14.0 (491537b, 2024-01-09T18:43:57Z)`

File: core/mondoo-aws-security.mql.yaml [hunk 2]
```diff
@@ -2265,7 +2265,7 @@ queries:
     filters: |
       asset.platform == "aws-sagemaker-notebookinstance"
     mql: |
-      aws.sagemaker.notebookInstance
+      aws.sagemaker.notebookInstances
```
Comment (vjeffrey) [inline core/mondoo-aws-security.mql.yaml:2268]: but don't we need the singular here, not the list of instances? i think it needs to be `notebookinstance`

Comment (mm-weber) [reply core/mondoo-aws-security.mql.yaml:2268]: Generally speaking, @tas50 original suggestion is the way to go.

```
aws.sagemaker.notebookinstance.details.kmsKey != empty
```

But if the singular resource doesn't [work](https://github.com/mondoohq/cnspec-policies/pull/339#discussion_r1462184734) [properly](https://github.com/mondoohq/cnquery/issues/3032), it is a good workaround to limit the number of assets on the general resource with a where-clause until it does work.

```
aws.sagemaker.notebookInstances
  .where(asset.name == name)
  .all(details.kmsKey != empty)
```

**Review by mm-weber — APPROVED** [review]
LGTM

File: core/mondoo-aws-security.mql.yaml (hunk 2 above)
Comment (vjeffrey) [reply core/mondoo-aws-security.mql.yaml:2268]: do we know the single one is not working properly? the example above doesn't show all lowercase notebookinstance, it's camel case

**Comment by vjeffrey** [comment]
ill see if i can get the fix for the resource in on friday, for now ya lets merge this

**Review by vjeffrey — APPROVED** [review]

**Pull request merged**